src/core/Typing.ml

```diff
@@ -833,7 +833,7 @@ let check_constr_boundary_sys cx dlbl desc sys =
         | `Same | `Indet ->
           begin
             try
-              let cx', _ = Cx.restrict cx r0 r1 in
+              let cx', phi = Cx.restrict cx r0 r1 in
```

Comment:
Did you use this `phi` anywhere?

Reply:
good catch.

By the way, i had disabled the 'unused variable as build error' thing, because it made it utterly impossible to debug `redtt` (debugging is a fancy word for commenting out lots of code, possibly making variables unused, and adding `Format.eprintf`).  But this has a negative effect on code clean-ness. I wish I knew how to make it a warning but not an error.